AVX512 not enabled with clang-8.0

See https://github.com/pitrou/arrow/runs/1972721611?check_suite_focus=true#step:8:900
```
In file included from /arrow/cpp/src/arrow/util/bpacking_avx512.cc:19:
/arrow/cpp/src/arrow/util/bpacking_simd512_generated.h:47:14: error: implicit instantiation of undefined template 'xsimd::batch<unsigned int, 16>'
  simd_batch masks(mask);
             ^
xsimd_ep/src/xsimd_ep-install/include/xsimd/types/xsimd_base.hpp:35:11: note: template is declared here
    class batch;
          ^
```

Yet the compilation command has the flags `-march=skylake-avx512 -mbmi2 -mavx512f -mavx512cd -mavx512vl -mavx512dq -mavx512bw`. Does xsimd expect more flags when detecting AVX512?

--- a/include/xsimd/config/xsimd_instruction_set.hpp
+++ b/include/xsimd/config/xsimd_instruction_set.hpp
@@ -106,7 +106,7 @@
 // AVX512 instructions are supported starting with gcc 6
 // see https://www.gnu.org/software/gcc/gcc-6/changes.html
 #if !defined(XSIMD_X86_INSTR_SET) && (defined(__AVX512__) || defined(__KNCNI__) || defined(__AVX512F__)\
-    && (!defined(__GNUC__) || __GNUC__ >= 6))
+    && (defined(__clang__) || (!defined(__GNUC__) || __GNUC__ >= 6)))
     #define XSIMD_X86_INSTR_SET XSIMD_X86_AVX512_VERSION
 
     #if defined(__AVX512VL__)

